src/watch.rs

```diff
@@ -35,22 +35,26 @@ impl WatchSystem {
         let (build_tx, build_rx) = channel(1);
 
         // Process ignore list.
-        let mut ignores = cfg.ignore.iter().try_fold(vec![], |mut acc, path| -> Result<Vec<PathBuf>> {
-            let abs_path = path.canonicalize().map_err(|err| anyhow!("invalid path provided: {}", err))?;
-            acc.push(abs_path);
-            Ok(acc)
-        })?;
-        ignores.append(&mut vec![cfg.build.dist.clone()]);
+        let mut ignored_paths =
+            cfg.ignored_paths
+                .iter()
+                .try_fold(Vec::with_capacity(cfg.ignored_paths.len() + 1), |mut acc, path| -> Result<Vec<PathBuf>> {
```

Comment:
Vec should be allocated with the capacity to prevent reallocation, the `+ 1` is for the `cfg.build.dist`